core/src/main/java/org/mule/DefaultMuleContext.java

```diff
@@ -323,6 +329,73 @@ public synchronized void start() throws MuleException
         }
     }
 
+    private void startMessageSources() throws LifecycleException
+    {
+        startPipelineMessageSources();
+        startServicesMessageSources();
+    }
+
+    private void startServicesMessageSources() throws LifecycleException
+    {
+        for (Service service : this.getRegistry().lookupObjects(Service.class))
+        {
+            if (service.isStarted() || service.isPaused())
+            {
+                MessageSource messageSource = service.getMessageSource();
+
+                startMessageSource(messageSource);
+
+                if (service.isPaused() && messageSource instanceof Pausable)
```

Comment:
There can be an small portion of time where the source would be started, may try to process messages but the service would be stopped. Maybe there is not way to solve this. WHat do you think?